src/v/test_utils/boost_result_redirect.cc

```diff
@@ -25,15 +25,16 @@ struct bazel_result_handler_for_boost {
     bazel_result_handler_for_boost() {
         using namespace boost::unit_test;
         if (auto xml_path = std::getenv("XML_OUTPUT_FILE")) {
-            _out = std::ofstream(xml_path);
-            // use fixed format to avoid scientific notation as a work-around
-            // to https://github.com/bazelbuild/bazel/issues/24605
-            _out << std::fixed << std::setprecision(6);
+            // use fixed format to avoid scientific notation as a
+            // work-around to
+            // https://github.com/bazelbuild/bazel/issues/24605
+            _out = std::ofstream(xml_path)
+                   << std::fixed << std::setprecision(6);
             unit_test_log.add_format(OF_JUNIT);
-            unit_test_log.set_stream(OF_JUNIT, _out);
+            unit_test_log.set_stream(OF_JUNIT, *_out);
         }
     }
-    std::ofstream _out;
+    static inline std::optional<std::ofstream> _out;
```

Comment:
Hmm, this was my doing. An example I saw had some deregister code in the desctructor which I omitted, which probably avoided the UAF.

So in this case `_out` outlives the problematic access? Because the config object itself is destroyed (I think on every test) but the the static outlives that, until static destruction time.

Reply:
> So in this case _out outlives the problematic access?

Yeah, the destructor for the unit test singleton runs before this from my testing and the sanitizer is happy too. I've not fully loaded the standard on destructor ordering into ram here, so this still could be sketchy. However it does seem the XML files are still being created properly, so I think we're good?

Reply:
> I think on every test

The documentation seems to disagree with that, I think everything we're playing with here are singletons, but the global configuration singleton is destroyed before the unit test stuff, which is why we need that to run.

https://live.boost.org/doc/libs/1_82_0/libs/test/doc/html/boost_test/utf_reference/link_references/link_boost_test_global_configuration.html